core/node/notifications/sync/streams_tracker_worker_connectgo.go

```diff
@@ -270,10 +270,15 @@ func (s *StreamTrackerConnectGo) Run(
 					continue
 				}
 
-				// apply update
 				for _, event := range update.GetStream().GetEvents() {
-					if err := trackedStream.HandleEvent(syncCtx, event); err != nil {
-						log.Error("Unable to handle event", "stream", streamID, "err", err)
+					if err := trackedStream.ApplyEvent(syncCtx, event); err != nil {
+						log.Error("Unable to apply event", "stream", streamID, "err", err)
+					}
+				}
+
+				for _, block := range update.GetStream().GetMiniblocks() {
+					if err := trackedStream.ApplyBlock(block, onChainConfig.Get()); err != nil {
+						log.Error("Unable to apply block", "stream", streamID, "err", err)
```

Comment:
seems like you should apply the blocks before the events... is there a reason to do it in the other order?

Reply:
My line of thinking was that events first enter the minipool and are dropped from the pool when included in a block. So the block is processed after the events are added to replicate this.